src/rvpt/rvpt.h

```diff
@@ -112,22 +112,15 @@ class RVPT
     std::vector<float> random_numbers;
 
     // BVH AABB's
-    BvhBuilder bvh_builder;
-    BvhNode* tl_bvh; // Highest level BVH
+    BinnedBvhBuilder bvh_builder;
+    Bvh top_level_bvh;
+
+    // Debug BVH view
     std::vector<std::vector<AABB>> depth_bvh_bounds;
     size_t bvh_vertex_count = 0;
-    int max_bvh_build_depth = 5;
-    int max_bvh_view_depth = 2;
+    int max_bvh_view_depth = 1;
     bool view_previous_depths = true;
-    bool does_bvh_support_depth_viewing = false;
-    // Yes I know, I have an extra set of the AABB's, I did this so instead of recursively going
```

Comment:
I took the liberty of removing that comment. You can add it back if you want. I think the current solution of having a copy of the AABBs makes sense. If you don't like that you can alternatively use `bvh::collect_aabbs_by_depth()` to reconstruct them.

Reply:
Oh yeah it's fine, 99% of this code written by me is really old and at this point I would write it completely differently anyways.